### ISSUE TITLE
Assignment 2 - Batchnorm 

### ISSUE THREAD

**Issue by LevyMatan** [hungduyle/CS231n-2018Spring#1]
Did you experienced any slow down after implementing the batch-norm while running a non-batch-norm solver?

I know this question might sound stupid, by I am currently struggling with it.
Thanks

**Comment by dhungle** [comment]
What level of slow down are your referring to? I didn't experience any thing significant in running time for me to notice. After seeing your question, I have also measured running time for BN and non BN solvers in the BatchNormalization.ipynb but nothing seemed too slow.

**Comment by LevyMatan** [comment]
Thanks, I am not sure what happened but it seemed to be fixed after a computer restart

**Comment by billlyzhaoyh** [comment]
@LevyMatan  
I encountered the same thing after writing in if statement switching to my custom training. The non-batch norm solver becomes much slower (an order of magnitude) smaller than the batch norm solver. I will see if restarting helps. When I am training the 8-layer neural net, it takes more than 2 hours to complete the weight scale experiment.
